wagtail/templatetags/wagtailcore_tags.py

```diff
@@ -98,7 +98,7 @@ def wagtail_release_notes_path():
 
 @register.simple_tag
 def wagtail_feature_release_whats_new_link():
-    return "https://guide.wagtail.org/en-latest/releases/new-in-wagtail-4-1/"
+    return "https://guide.wagtail.org/en-latest/releases/latest"
```

Comment:
Let's prevent an extra redirect here

```suggestion
    return "https://guide.wagtail.org/en-latest/releases/latest/"
```

Reply:
Also need to change here:

https://github.com/wagtail/wagtail/blob/667379f9ff9f202e242fb9ea935c48e276b60050/wagtail/admin/wagtail_hooks.py#L976-L977

Not sure if we should manually update it to 5.0 now, or make use of `VERSION` in `wagtail/__init__.py`